test/e2e/clusters/clusters.go

```diff
@@ -0,0 +1,256 @@
+	DistroK3s
+
+	kindClusterName = "kind-zarf-test"
+	k3dClusterName  = "k3d-zarf-test"
+)
+
+// GetDistroToUseFromString decides which cluster the user wants based on the string value passed from an environment
+// variable.
+func GetDistroToUseFromString(s string) (DistroToUse, error) {
+	match := map[string]DistroToUse{
+		"provided": DistroProvided,
+		"kind":     DistroKind,
+		"k3d":      DistroK3d,
+		"k3s":      DistroK3s,
+	}
+	if distroToUse, ok := match[s]; ok {
+		return distroToUse, nil
+	} else {
+		return DistroUnknown, fmt.Errorf("\"%v\" is not a valid value for determining which k8s distro to use", s)
+	}
+}
+
+// CreateClusterWithTemporaryKubeconfig creates a temporary Kubeconfig file, updates the process's KUBECONFIG env var
+// to point at the new file, and creates a kubernetes cluster. It returns the path to the temporary Kubeconfig file,
+// or an error if something went wrong. The caller is responsible for destroying the cluster and deleting the temporary
+// kubeconfig file. The KUBECONFIG env var doesn't need to be reset since it was only changed for the current process.
+func CreateClusterWithTemporaryKubeconfig(distroToUse DistroToUse) (string, error) {
+	// Create the temporary Kubeconfig file
+	var clusterName string
+	switch distroToUse {
+	case DistroKind:
+		clusterName = kindClusterName
+	case DistroK3d:
+		clusterName = k3dClusterName
+	}
+	tempKubeconfigFile, err := ioutil.TempFile("", clusterName+"*.yaml")
+	if err != nil {
+		return "", err
+	}
+
+	// Set the KUBECONFIG env var to the new file
+	err = os.Setenv("KUBECONFIG", tempKubeconfigFile.Name())
+	if err != nil {
+		return "", err
+	}
+
+	// Create the cluster
+	switch distroToUse {
+	case DistroKind:
+		err = createKindCluster(tempKubeconfigFile.Name())
+		if err != nil {
+			return "", err
+		}
+	case DistroK3d:
+		err = createK3dClusterUsingCurrentKubeconfig()
+		if err != nil {
+			return "", err
+		}
+	}
+	return tempKubeconfigFile.Name(), nil
+}
+
+// DeleteClusterAndTemporaryKubeconfig is intended to be deferred at the end of TestMain execution so we can clean up
+// the cluster and temp file that we made.
+func DeleteClusterAndTemporaryKubeconfig(distroToUse DistroToUse, tempKubeconfigFilePath string) error {
+	// Defer the temp file delete so it always happens no matter what
+	defer func(name string) {
+		_ = os.Remove(name)
+	}(tempKubeconfigFilePath)
+
+	// Delete the cluster
+	switch distroToUse {
+	case DistroKind:
+		err := deleteKindCluster(tempKubeconfigFilePath)
+		if err != nil {
+			return err
+		}
+	case DistroK3d:
+		err := deleteK3dCluster()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// TryValidateClusterIsRunning establishes a valid connection to the currently configured cluster, then returns,
+// throwing an error if anything went wrong.
+func TryValidateClusterIsRunning() error {
+	clientSet, err := GetClientSet()
```

Comment:
Why are we implementing custom logic to do what [common.go](https://github.com/defenseunicorns/zarf/blob/master/cli/internal/k8s/common.go#L104) already does?

Reply:
This was moved from a different an existing function in the test code that existed before the new cli-supported common.go function. You're right that we should be using that instead now that it exists though.

Reply:
The custom logic was already there before this PR. I can look at pulling it out.

Reply:
Yeah let's use the k8s package for this, thx

Reply:
Do we want to move `k8s` out of the `internal` folder so we can do this? That's a bigger decision/change than I'm willing to take on in this PR.

Reply:
It's all one go.mod, would we need to move it out of internal?  I think _we should move it_, but agree that's out of scope for this PR.

Reply:
<img width="1325" alt="image" src="https://user-images.githubusercontent.com/16000938/159347374-49c03786-c06e-499e-9e8c-b4d20fe9147a.png">

Might just be an IDE thing, but still not something I want to do even if it is technically possible.

Reply:
I suggest we stick with what's there for now and updated it as part of #411

Reply:
@jeff-mccoy are you good with that?